source/patterns/@aws-solutions-constructs/aws-route53-apigateway/README.md

```diff
@@ -0,0 +1,122 @@
+    endpointConfiguration: {
+        types: [api.EndpointType.REGIONAL]
+    },
+    defaultMethodOptions: {
+        authorizationType: api.AuthorizationType.NONE
+    }
+};
+
+const apiGateway = new api.LambdaRestApi(this, 'LambdaRestApi', apiGatewayProps);
+
+const parentZone = new route53.PublicHostedZone(this, 'HostedZone', {
+  zoneName: 'example.com',
+});
+
+// This construct can only be attached to a configured API Gateway.
+new Route53ToApigateway(this, 'Route53ToApigatewayPattern', {
+    existingApiGatewayObj: apiGateway,
+    existingHostedZoneInterface: parentZone,
+    publicApi: true
+});
+
+```
+
+## Initializer
+
+``` text
+new Route53ToApigateway(scope: Construct, id: string, props: Route53ToApigatewayProps);
+```
+
+_Parameters_
+
+* scope [`Construct`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html)
+* id `string`
+* props [`Route53ToApigatewayProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+This construct cannot create a new Public Hosted Zone, if you are creating a public API you must supply an existing Public Hosted Zone that will be reconfigured with a new Alias record. Public Hosted Zones are configured with public domain names and are not well suited to be launched and torn down dynamically, so this construct will only reconfigure existing Public Hosted Zones.
+
+This construct can create Private Hosted Zones. If you want a Private Hosted Zone, then you can either provide an existing Private Hosted Zone or a privateHostedZoneProps value with at least the Domain Name defined.
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+| privateHostedZoneProps? | [route53.PrivateHostedZoneProps](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.PrivateHostedZoneProps.html) | Optional custom properties for a new Private Hosted Zone. Cannot be specified for a public API. Cannot specify a VPC, it will use the VPC in existingVpc or the VPC created by the construct. Providing both this and existingHostedZoneInterface is an error. |
+| existingHostedZoneInterface? | [route53.IHostedZone](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.IHostedZone.html) | Existing Public or Private Hosted Zone (type must match publicApi setting). Specifying both this and privateHostedZoneProps is an error. If this is a Private Hosted Zone, the associated VPC must be provided as the existingVpc property.|
+| publicApi | boolean | Whether the construct is deploying a private or public API. This has implications for the Hosted Zone and VPC. |
+| vpcProps? | [ec2.VpcProps](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.VpcProps.html) | Optional custom properties for a VPC the construct will create. This VPC will be used by any Private Hosted Zone the construct creates (The privateHostedZoneProps cannot include a VPC). Providing both this and existingVpc is an error. |
```

Comment:
Interesting - this goes back to my previous comment. In any event, a private API will already define the VPC and we only accept existing APIs, so this construct will never create a VPC and we can drop this prop. If someone brings a private API they created via CDK, we still want them to give us the VPC in existing VPC.